utils/scripts/nf/__init__.py

```diff
@@ -150,6 +150,15 @@ def docker_registry(self) -> str:
             raise fnfe
         return _docker_registry
         
+    def is_ecr_private_repo(self, _registry_name):
+        """
+        Detect private registry in format <awsaccountid>.dkr.ecr.<awsregion>.amazonaws.com
+        """
+        tmp = _registry_name.split('.')
+        if len(tmp) == 6 and tmp[1] == 'dkr' and tmp[2] == 'ecr' and tmp[4] == 'amazonaws' and tmp[5] == 'com':
```

Comment:
Nit: This conditional can be the return statement as it will provide the boolean value needed, e.g:

```python
return len(tmp) == 6 and tmp[1] == 'dkr' and tmp[2] == 'ecr' and tmp[4] == 'amazonaws' and tmp[5] == 'com'
```

Alternatively, you can use a regex pattern match:
```python
return re.match('[0-9]{12}\\.dkr\\.ecr\\.amazonaws\\.com', _registry_name)
```